backend-services/autostarter/core/autostarter.py

```diff
@@ -0,0 +1,300 @@
+import os
+import re
+import socket
+import time
+
+import requests
+import ujson as json
+from artemis_utils import AUTO_RECOVER_PROCESS_STATE
+from artemis_utils import DB_HOST
+from artemis_utils import DB_NAME
+from artemis_utils import DB_PASS
+from artemis_utils import DB_PORT
+from artemis_utils import DB_USER
+from artemis_utils import get_logger
+from artemis_utils import TEST_ENV
+from artemis_utils.db_util import DB
+
+# logger
+log = get_logger()
+
+# global vars
+CHECK_INTERVAL = int(os.getenv("CHECK_INTERVAL", 5))
+SERVICE_NAME = "autostarter"
+AUTOIGNORE_HOST = "autoignore"
+CONFIGURATION_HOST = "configuration"
+DATABASE_HOST = "database"
+FILEOBSERVER_HOST = "fileobserver"
+PREFIXTREE_HOST = "prefixtree"
+NOTIFIER_HOST = "notifier"
+DETECTION_HOST = "detection"
+MITIGATION_HOST = "mitigation"
+RIPERISTAP_HOST = "riperistap"
+BGPSTREAMLIVETAP_HOST = "bgpstreamlivetap"
+BGPSTREAMKAFKATAP_HOST = "bgpstreamkafkatap"
+BGPSTREAMHISTTAP_HOST = "bgpstreamhisttap"
+EXABGPTAP_HOST = "exabgptap"
+REST_PORT = int(os.getenv("REST_PORT", 3000))
+ALWAYS_RUNNING_SERVICES = [
+    CONFIGURATION_HOST,
+    DATABASE_HOST,
+    NOTIFIER_HOST,
+    FILEOBSERVER_HOST,
+    PREFIXTREE_HOST,
+    AUTOIGNORE_HOST,
+]
+USER_CONTROLLED_SERVICES = [
+    DETECTION_HOST,
+    MITIGATION_HOST,
+    RIPERISTAP_HOST,
+    BGPSTREAMLIVETAP_HOST,
+    BGPSTREAMKAFKATAP_HOST,
+    BGPSTREAMHISTTAP_HOST,
+    EXABGPTAP_HOST,
+]
+
+# trigger queries
+DROP_TRIGGER_QUERY = "DROP TRIGGER IF EXISTS send_update_event ON public.bgp_updates;"
+CREATE_TRIGGER_QUERY = "CREATE TRIGGER send_update_event AFTER INSERT ON bgp_updates FOR EACH ROW EXECUTE PROCEDURE rabbitmq.on_row_change('update-insert');"
+# TODO: move to utils
+IS_KUBERNETES = os.getenv("KUBERNETES_SERVICE_HOST") is not None
+
+
+# TODO: move to utils
+def service_to_ips_and_replicas(base_service_name):
+    service_to_ips_and_replicas_set = set([])
+    addr_infos = socket.getaddrinfo(base_service_name, REST_PORT)
+    for addr_info in addr_infos:
+        af, sock_type, proto, canon_name, sa = addr_info
+        replica_ip = sa[0]
+        replica_host_by_addr = socket.gethostbyaddr(replica_ip)[0]
+        replica_name_match = re.match(
+            r"^artemis_" + re.escape(base_service_name) + r"_(\d+)",
```

Comment:
the prefix is based on the folder name we run docker-compose in so it's not always artemis. We should remove the prefix from the regex. Same applies for this function on the other modules.

Reply:
OK I will fix and commit

Reply:
Note that this is taken by: `COMPOSE_PROJECT_NAME=artemis`
I will adjust the prefix so that it actually uses this env var (with artemis as default)

Reply:
Fixed in https://github.com/FORTH-ICS-INSPIRE/artemis/pull/472/commits/158d3cf86d8f89285dfd64171520ecb37ff119cb please check